docs/_v15_x/4_archive.md

```diff
@@ -74,11 +74,59 @@ Archive current = archives.getArchives().get(0);
 final List<ArchiveDocument> documents = new ArrayList<>();
 
 while (current.getNextDocuments().isPresent()) {
-current = current.getNextDocuments()
-.map(client::getArchiveDocuments)
-.orElse(new Archive());
+    current = current.getNextDocuments()
+    .map(client::getArchiveDocuments)
+    .orElse(new Archive());
 
-documents.addAll(current.getDocuments());
+    documents.addAll(current.getDocuments());
+}
+
+// This prints to total content of the list of documents
+System.out.println(documents);
+```
+## Archive Document attributes
+
+You can add optional attributes to documents. An attribute is a key/val string-map that describe documents. You can add 
+up to 15 attributes pr. archive document. 
+
+```java
+final ArchiveDocument invoice = new ArchiveDocument(
+    UUID.randomUUID()
+    , "invoice_123123.pdf"
+    , "pdf"
+    , "application/pdf"
+).withAttribute("INR", "123123").withAttribute("custid", "4321");
+```
+
+The attributes can be queried, so that you can get an iterable list of documents.
+
+```java
+final Archives archives = digipostClient.getArchives();
+Archive current = archives.getArchives().get(0);
+
+final List<ArchiveDocument> documents = current.getNextDocumentsWithAttributes(Map.of("INR", "123123", "custid", "4321"))
+        .map(digipostClient::getArchiveDocuments)
+        .map(Archive::getDocuments).orElse(emptyList());
+
+// This prints to total content of the list of documents
```

Comment:
Forstår ikkje heilt denne kommentaren ("to total content"). Den printer alt?

Reply:
Ja, det er vel en rekke med toString-metoder på alle objektene i listen slik at du skriver ut til skjerm.